### PR TITLE
Added OptDomainNameServer for DHCPv4

### PR DESCRIPTION
**Pull request by insomniacslk**
Added option decoder for the Domain Name Server option in DHCPv4

(tests coming in the next commit)

### CODE REVIEW

File: dhcpv4/option_domain_name_server.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+import (
+	"fmt"
+	"net"
+)
+
+// This option implements the domain name server option
+// https://tools.ietf.org/html/rfc2132
+
+// OptDomainNameServer represents an option encapsulating the server identifier.
```
Comment (pmazzini) [inline dhcpv4/option_domain_name_server.go]: server identifier -> domain name servers

File: dhcpv4/option_domain_name_server.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+func ParseOptDomainNameServer(data []byte) (*OptDomainNameServer, error) {
+	if len(data) < 2 {
+		return nil, ErrShortByteStream
+	}
+	code := OptionCode(data[0])
+	if code != OptionDomainNameServer {
+		return nil, fmt.Errorf("expected code %v, got %v", OptionDomainNameServer, code)
+	}
+	length := int(data[1])
+	if length%4 != 0 {
```
Comment (pmazzini) [inline dhcpv4/option_domain_name_server.go]: I think we should also check if the lenght is > 0, the minimun length should be 4.

File: dhcpv4/option_domain_name_server.go (hunk 1 above)
Comment (insomniacslk) [reply dhcpv4/option_domain_name_server.go]: silly me. Good catch! I used another option file as template and didn't replace everything

File: dhcpv4/option_domain_name_server.go (hunk 2 above)
Comment (insomniacslk) [reply dhcpv4/option_domain_name_server.go]: you are right - according to RFC2132 par3.8 the minimum length has to be four octets (one NS)

**Comment by insomniacslk** [comment]
@pmazzini can you reapprove?

**Pull request merged**